src/twisted/internet/_sslverify.py

```diff
@@ -1703,6 +1707,14 @@ def _verifyCallback(conn, cert, errno, depth, preverify_ok):
             ctx.load_tmp_dh(self.dhParameters._dhFile.path)
         ctx.set_cipher_list(self._cipherString.encode('ascii'))
 
+        # OpenSSL 1.0.2 introduced the SSL_CTX_set_ecdh_auto so that you don't
+        # need to set a supported key. OpenSSL 1.1.0 removed the function
+        # again and it's always enabled.
+        try:
+            self._lib.SSL_CTX_set_ecdh_auto(ctx, 1);
+        except BaseException:
```

Comment:
I think that here we should have an explicit exception.
I am looking at his code, but I have no idea where the `self._lib` came from.

Maybe this should be `OpenSSL._util.lib.SSL_CTX_set_ecdh_auto` ?

I think that we need to new test to check that on other systems it raises the expected exception.

I am running some tests for 

```
pyOpenSSL OpenSSL 0.9.8j-fips 07 Jan 2009 (9470127)
pyOpenSSL 16.0.0
```

and various other systems and will report the results, but that self._lib is strange

Reply:
I have tried on 1.1.0 and the method is still there, so the comment is misleading :)

```
pyOpenSSL OpenSSL 1.1.0f  25 May 2017 (269484143)
pyOpenSSL 17.0.0

>>> import OpenSSL
>>> import OpenSSL.SSL
>>> OpenSSL.SSL.OPENSSL_VERSION_NUMBER
269484143
>>> OpenSSL._util.lib.SSL_CTX_set_ecdh_auto
<built-in method SSL_CTX_set_ecdh_auto of CompiledLib object at 0x7f7705e34c58>

Reply:
OpenSSL 1.1.0 changed SSL_CTX_set_ecdh_auto to a define that does nothing. It always was a define, but in 1.0.2 it called SSL_CTX_ctrl().

Reply:
The self._lib thing comes from the _OpenSSLECCurve class, I just copied it not understanding what I was doing at the time. I also pointed that out that I had no idea what I was doing with regard to this function when I created the merge request.